thriftbp/prometheus.go

```diff
@@ -255,3 +255,12 @@ func stringifyErrorType(err error) string {
 	}
 	return strings.TrimPrefix(fmt.Sprintf("%T", err), "*")
 }
+
+// We create this interface instead of using errors.go/Error because we want it
+//to always match the Baseplate.Error regardless of the version the user has put in their repository.
+type baseplateErrorMinimalIface interface {
```

Comment:
`minimalBaseplateError` then in case we later add the retriable.

Reply:
I actually slightly prefer `baseplateErrorCode` between the 2 names I suggested, because this is more specific (we only need code for prometheus purpose, so the name reflects that, while the term "minimal" is quite ambiguous).

Reply:
agreed, updating again